lua/orgmode/api/init.lua

```diff
@@ -101,7 +100,67 @@ function OrgApi.refile(opts)
   return true
 end
 
+--- Get a link destination as string
+---
+--- Depending if org_id_link_to_org_use_id is set the format is
+---
+--- id:<uuid>::*title and the id is created if not existing
+--- or
+--- file:<filepath>::*title
+---
+--- The result is meant to be used as link_location for OrgApi.insert_link.
+--- @param headline OrgApiHeadline
+--- @return string
+function OrgApi.get_link_to_headline(headline)
+  local filename = headline.file.filename
+  local bufnr = vim.fn.bufnr(filename)
+
+  if bufnr == -1 then
```

Comment:
Why do we need to do this if there is no buffer?

Reply:
Jep, actually this is the crucial part (and took me some time to figure out, reading through the agenda code). 

The file we want to target a link to, is very likely not open. If we use `org_id_link_to_org_use_id = true`, there might not be an ID yet. But creating this id needs an edit of the file, which is done by the `id_get_or_create` methods under the hood. And this needs to open the file automatically in the background and close it again, if it is not already open.

Let me clarify the motivation behind this API extension:
When I use telescope-orgmode to create links, I can access the whole index of files build up by orgmode on startup. This allows for very convenient creation of links and cross-references using fuzzy finding. It is not very convenient, if I would have to open the files first, before being able to set the link.
The manual workflow for ID links works actually that way, I have to open the target file, press `org-store-link` (which also creates the id), navigate to my file, where the link originates and press `org-insert-link`. My telescope plugin makes this workflow much smoother, because I can just filter for the target headline without open the file, press enter and, boom, I have the link and the ID.

Reply:
I think, we should keep it as it is for know. I expect to discover more requirements while I am implementing more features and playing around with we have. If it doesn't bother you, I would adapt when need and have better knowledge.

Reply:
I just figured out I didn't phrase my question correctly.

What I wanted to ask is:
Would it work for you to move these to api/file and api/headline? All of the file or headline specific api methods live in those file, so I'd prefer to keep it consistent.

If it doesn't work for you, can you clarify why?

Reply:
All three methods are used for link creation and are meant to be used together. I might need to understand better, what your primary criterion is for where to put a method and also what the main purpose of the overall API is.
I don't actually have a big problem of making these methods member functions of `OrgApiFile` and `OrgApiHeadline`.

Reply:
>  I might need to understand better, what your primary criterion is for where to put a method

If a function is accepting an instance of something, I always prefer to have that function on the instance itself. Of course that's not always possible, but when it is, I prefer to take that path.
For example, if we would follow this way for example for `get_property`, we would have something like this:

```lua
orgmode.api.get_property(headline, 'id')
```

But this is much cleaner IMO:

```lua
headline:get_property('id')
```

Reply:
Yes, was already convienced, when I realized, that the argument of the original functions is actually the self argument in `orgmode.api.headline` and `orgmode.api.file`. . :smile: 
Now that I refactored the code I am even more convienced, because it cleaned also things up on the client side.

What do you think, are we ready to merge?